pkg/util/storage/manager.go

```diff
@@ -44,6 +47,20 @@ func (m *manager) BlobService(ctx context.Context, resourceGroup, account string
 		SharedAccessExpiryTime: &date.Time{Time: t.Add(24 * time.Hour)},
 	})
 	if err != nil {
+		if strings.Contains(err.Error(), "TooManyRequests") {
```

Comment:
Why match on a sting in the error message and not the `res.Response.StatusCode`?

Reply:
I dont think there is much impact when it is changed to `if err.(autorest.DetailedError).StatusCode == "TooManyRequests"`

Reply:
I was thinking more about matching against the number. Like we do here: https://github.com/azure/ARO-RP/blob/7cbf95b811d79677594809f90c8e45681a382b49/pkg/database/cosmosdb/zz_generated_cosmosdb.go#L73

This way if they change "TooManyRequest" -> "Too Many Requests" or whatever we're still good. IMO Matching on the statusCode is more robust. 

What do you think?

Reply:
@mociarain : could you please review the code now? 
let me know if this is more convincing or not

Reply:
That's exactly what I had in mind